docs/_layouts/default.html

```diff
@@ -90,6 +90,32 @@
   <div id="fb-root"></div>
 
   <script>
+    (function() {
+      var selector = '.inner-content h2, .inner-content h3, .inner-content h4';
+      var elements = document.querySelectorAll(selector);
+      for (var i = 0; i < elements.length; i++) {
+        var elementID;
+
+        if (elements[i].hasAttribute('id')) {
+          elementID = elements[i].getAttribute('id');
+        } else {
+          var textMethod = document.body.textContent ? "textContent" : "innerText";
+          var roughText = elements[i][textMethod];
+
+          // Regex rule for making the title URL-friendly.
+          tidyText = roughText.toLowerCase()
+                              .replace(/\s+/g, '-')
+                              .replace(/[^A-Za-z0-9\-_.]/g, '');
+
+          elements[i].setAttribute('id', tidyText);
+          elementID = tidyText;
+        }
+        var anchor = '<a class="anchor" href="#' + elementID + '">#</a>';
+        elements[i].innerHTML = elements[i].innerHTML + anchor;
```

Comment:
This is really inefficient as it needs to serialize the HTML (the read of `innerHTML`), then append text, then deserialize the text back into DOM nodes. I'd suggest creating a DOM node directly instead, something like this (untested):
```
var anchor = document.createElement('a');
anchor.className = 'anchor';
anchor.href = '#' + elementID;
elements[i].appendChild(anchor);
```